Scripts/Source/UDItemManager.psc

```diff
@@ -13,31 +14,35 @@ Event OnInit()
     endif
 EndEvent
 
+; TODO
+; boots high heels
+; boots pony
+
 Function lockAbadonPiercings(Actor akTarget)
-    if !akTarget.wornhaskeyword(libs.zad_DeviousPiercingsVaginal)
+    if !akTarget.wornhaskeyword(libs.zad_DeviousPiercingsVaginal) && UDmain.UDRRM.IsDeviceFiltered(0)
         libs.LockDevice(akTarget,UDmain.UDlibs.AbadonPiercingVaginal)
     endif
-    if !akTarget.wornhaskeyword(libs.zad_DeviousPiercingsNipple)
+    if !akTarget.wornhaskeyword(libs.zad_DeviousPiercingsNipple) && UDmain.UDRRM.IsDeviceFiltered(1)
         libs.LockDevice(akTarget,UDmain.UDlibs.AbadonPiercingNipple)
     endif
 EndFunction
 
 Function lockAbadonHelperPlugs(Actor akTarget)
-    if !akTarget.wornhaskeyword(libs.zad_deviousPlugAnal) && !AbadonScript.UseAnalVariant
-        if Utility.randomInt(0,1)
+    if !akTarget.wornhaskeyword(libs.zad_deviousPlugAnal) && !AbadonScript.UseAnalVariant && UDmain.UDRRM.IsDeviceFiltered(3)
+        if RandomInt(0,1)
```

Comment:
Maybe add Abadon Quest completion here, thus allowing random plugs without risk of them evolving into Abadon Plug?
Something like if RandomInt(0,1) && AbadonQuest.IsComplete()?

Reply:
Good point. Will do